Name: alarm daily options
Version: 1.3.1-1
Author: b romano
Description: This patch will add the daily values Sunday, Monday, Tuesday, Wednesday, etc. to your alarm options.

--- .orig/usr/palm/applications/com.palm.app.clock/app/models/alarms.js
+++ /usr/palm/applications/com.palm.app.clock/app/models/alarms.js
@@ -134,5 +134,12 @@
 	"daily": $L("Daily"),
 	"once": $L("     "), 
 	"weekdays": $L("Weekdays"),
-	"weekends": $L("Weekends")
+	"weekends": $L("Weekends"),
+	"sunday": $L("Sundays"),
+	"monday": $L("Mondays"),
+	"tuesday": $L("Tuesdays"),
+	"wednesday": $L("Wednesdays"),
+	"thursday": $L("Thursdays"),
+	"friday": $L("Fridays"),
+	"saturday": $L("Saturdays")
 }
--- .orig/usr/palm/applications/com.palm.app.clock/app/controllers/alarm-assistant.js
+++ /usr/palm/applications/com.palm.app.clock/app/controllers/alarm-assistant.js
@@ -5,7 +5,14 @@
 	    {label:$L('Ring Once'), value:"once"},
 		{label:$L('Daily'), value:"daily" },
 		{label:$L('Weekdays'), value:"weekdays"}, 
-		{label:$L('Weekends'), value:"weekends"} 
+		{label:$L('Weekends'), value:"weekends"},
+		{label:$L('Sundays'), value:"sunday"},
+		{label:$L('Mondays'), value:"monday"},
+		{label:$L('Tuesdays'), value:"tuesday"},
+		{label:$L('Wednesdays'), value:"wednesday"},
+		{label:$L('Thursdays'), value:"thursday"},
+		{label:$L('Fridays'), value:"friday"},
+		{label:$L('Saturdays'), value:"saturday"}
 	],
 	initialize: function(alarm, deleteSelf) {
 		this.appControl = Mojo.Controller.getAppController();
--- .orig/usr/palm/applications/com.palm.app.clock/app/models/alarm.js
+++ /usr/palm/applications/com.palm.app.clock/app/models/alarm.js
@@ -232,6 +232,27 @@
 			case ('weekends'):
 				now = this.nextWeekend(d,now);
 				break;
+                        case ('sunday'):
+				now = this.nextSpecificDay(d,now,Alarm.kSunday);
+				break;
+			case ('monday'):
+				now = this.nextSpecificDay(d,now,Alarm.kMonday);
+				break;
+			case ('tuesday'):
+				now = this.nextSpecificDay(d,now,Alarm.kTuesday);
+			        break;
+			case ('wednesday'):
+			        now = this.nextSpecificDay(d,now,Alarm.kWednesday);
+			        break;
+			case ('thursday'):
+                                now = this.nextSpecificDay(d,now,Alarm.kThursday);
+                                break;
+			case ('friday'):
+                                now = this.nextSpecificDay(d,now,Alarm.kFriday);
+                                break;
+                        case ('saturday'):
+				now = this.nextSpecificDay(d,now,Alarm.kSaturday);
+				break;
 			default:
 				now = this.nextDay(d,now);
 				break;
@@ -240,6 +261,30 @@
 		return now;
 		
 	},
+
+	nextSpecificDay: function(d,now,day){
+                var timePassed = false;
+		if (now.getHours() > d.getHours() || (now.getHours() == d.getHours() && now.getMinutes() >= d.getMinutes())) {
+			timePassed = true;
+		}
+       
+		if (day > today){
+			now.setDate(now.getDate() + (day - today));
+		}
+		else if (day == today){
+			if(timePassed){
+				now.setDate(now.getDate()+7);
+			}
+		}
+		else if (day < today){
+			now.setDate(now.getDate() + (7 + day - today));
+		}
+
+		now.setHours(d.getHours());
+		now.setMinutes(d.getMinutes());
+		now.setSeconds(0);
+		return now;
+	},
 	
 	nextDay: function(d,now){
 		if (now.getHours() > d.getHours() || (now.getHours() == d.getHours() && now.getMinutes() >= d.getMinutes())) {
@@ -332,6 +377,76 @@
 				} else {
 					return (today == Alarm.kSunday ? 6 : 1)
 				}
+			case 'sunday':
+				if (today == Alarm.kSunday && this.isLaterToday()) {
+					return 0;
+				}
+				else if (today >= Alarm.kSunday){
+					return (7 + Alarm.kSunday - today);
+				}
+				else if (today < Alarm.kSunday){
+					return Alarm.kSunday - today;
+				}
+			case 'monday':
+				if (today == Alarm.kMonday && this.isLaterToday()) {
+				        return 0;
+				}
+				else if (today >= Alarm.kMonday){
+					return (7 + Alarm.kMonday - today);
+				} 
+				else if (today < Alarm.kMonday){
+					return Alarm.kMonday - today;
+				}
+			case 'tuesday':
+				if (today == Alarm.kTuesday && this.isLaterToday()) {
+					return 0;
+				}
+				else if (today >= Alarm.kTuesday){
+					return (7 + Alarm.kTuesday - today);
+				}
+				else if (today < Alarm.kTuesday){
+					return Alarm.kTuesday - today;
+				}
+			case 'wednesday':
+				if (today == Alarm.kWednesday && this.isLaterToday()) {
+					return 0;
+				}
+				else if (today >= Alarm.kWednesday){
+					return (7 + Alarm.kWednesday - today);
+				}
+				else if (today < Alarm.kWednesday){
+					return Alarm.kWednesday - today;
+				}
+			case 'thursday':
+				if (today == Alarm.kThursday && this.isLaterToday()) {
+					return 0;
+				}
+				else if (today >= Alarm.kThursday){
+					return (7 + Alarm.kThursday - today);
+				}
+				else if (today < Alarm.kThursday){
+					return Alarm.kThursday - today;
+				}
+			case 'friday':
+				if (today == Alarm.kFriday && this.isLaterToday()) {
+					return 0;
+				}
+				else if (today >= Alarm.kFriday){
+					return (7 + Alarm.kFriday - today);
+				}
+				else if (today < Alarm.kFriday){
+					return Alarm.kFriday - today;
+				}
+			case 'saturday':
+				if (today == Alarm.kSaturday && this.isLaterToday()) {
+					return 0;
+				}
+				else if (today >= Alarm.kSaturday){
+					return (7 + Alarm.kSaturday - today);
+				}
+				else if (today < Alarm.kSaturday){
+					return Alarm.kSaturday - today;
+				}
 			// if it rings daily or once, then it either rings today or tomorrow
 			case 'daily':
 			case 'once':
@@ -423,6 +538,76 @@
 				} else {
 					return "";
 				}
+			case 'sunday':
+				if (today == Alarm.kSunday && this.isLaterToday()) {
+					return "";
+				}
+				else if (today == Alarm.kSaturday){
+					return $L("tomorrow");
+				}
+				else {
+					return $L("Sunday");
+				}
+			case 'monday':
+				if (today == Alarm.kMonday && this.isLaterToday()) {
+					return "";
+				}
+				else if (today == Alarm.kSunday){
+					return $L("tomorrow");
+				}
+				else {
+					return $L("Monday");
+				}
+			case 'tuesday':
+				if (today == Alarm.kTuesday && this.isLaterToday()) {
+					return "";
+				}
+				 else if (today == Alarm.kMonday){
+				 	return $L("tomorrow");
+				 }
+				 else {
+				 	return $L("Tuesday");
+				 }
+			case 'wednesday':
+				if (today == Alarm.kWednesday && this.isLaterToday()) {
+					return "";
+				}
+				else if (today == Alarm.kTuesday){
+					return $L("tomorrow");
+				}
+				else {
+					return $L("Wednesday");
+				}
+			case 'thursday':
+				if (today == Alarm.kThursday && this.isLaterToday()) {
+					return "";
+				}
+				else if (today == Alarm.kWednesday){
+					return $L("tomorrow");
+				}
+				else {
+					return $L("Thursday");
+				}
+			case 'friday':
+				if (today == Alarm.kFriday && this.isLaterToday()) {
+					return "";
+				}
+				else if (today == Alarm.kThursday){
+					return $L("tomorrow");
+				}
+				else {
+					return $L("Friday");
+				}
+			case 'saturday':
+				if (today == Alarm.kSaturday && this.isLaterToday()) {
+					return "";
+				}
+				else if (today == Alarm.kFriday){
+					return $L("tomorrow");
+				}
+				else {
+					return $L("Saturday");
+				}
 			case 'daily':
 			case 'once':
 				if (this.isLaterToday()) {
